### PR TITLE
Replace libsignal-protocol-c with libsodium

### PR DESCRIPTION
**Pull request by jagerman**
This uses a custom build of libsodium (https://github.com/jagerman/libsodium-internals) that exposes the internal `fe25519_*` functions that we need to properly do the signing/verification/pubkey conversions.

This also gets rid of the stupid 256-byte length limitation.

### CODE REVIEW

File: onionreq/signal-xed25519.cpp [hunk 1]
```diff
@@ -2,46 +2,141 @@
 
 #include "signal-xed25519.hpp"
 #include <cassert>
+#include <cstring>
 #include <stdexcept>
 #include <sodium/randombytes.h>
+#include <sodium/crypto_internal_fe25519.h>
```
Comment (majestrate) [inline onionreq/signal-xed25519.cpp:8]: if sodium headers are in `/usr/include` how will this deal with that ?

File: onionreq/signal-xed25519.cpp [hunk 2]
```diff
@@ -2,46 +2,141 @@
...
+        x[i] = f[i] ^ g[i];
+    unsigned char mask = (unsigned char) (-(signed char) b);
+    for (size_t i = 0; i < x.size(); i++)
+        x[i] &= mask;
+    for (size_t i = 0; i < x.size(); i++)
+        f[i] ^= x[i];
 }
 
-namespace onionreq::xed25519 {
+void fe25519_montx_to_edy(fe25519 y, const fe25519 u) {
```
Comment (majestrate) [inline onionreq/signal-xed25519.cpp:34]: `fe25519 & y` ?

Comment (jagerman) [reply onionreq/signal-xed25519.cpp:34]: fe25519 is typedef'ed to either a uint64_t[5] or a uint32_t[10], depending on the architecture, but both get passed as pointers.

Comment (majestrate) [reply onionreq/signal-xed25519.cpp:34]: ew

File: onionreq/signal-xed25519.cpp (hunk 1 above)
Comment (majestrate) [reply onionreq/signal-xed25519.cpp:8]: nevermind, `BEFORE SYSTEM`

**Pull request merged**